auth/frontend/src/main/scala/io/udash/auth/AuthPresenter.scala

```diff
@@ -0,0 +1,19 @@
+package io.udash.auth
+
+import io.udash._
+
+/**
+  * Presenter which check user access in `handleState` method.
+  *
+  * @param permission PermissionCombinator verified against provided `userCtx`.
+  * @param requireAuthenticated If `true`, the presenter requires `userCtx` to don't be [[io.udash.auth.UserCtx.Unauthenticated]] subclass.
+  */
+abstract class AuthPresenter[S <: State](permission: PermissionCombinator, requireAuthenticated: Boolean = false)
+                                        (implicit userCtx: UserCtx) extends Presenter[S] {
+
+  override def handleState(state: S): Unit = {
+    if (requireAuthenticated && userCtx.isInstanceOf[UserCtx.Unauthenticated]) throw new UnauthenticatedException()
```

Comment:
If we design user authentication through different context classes but we don't intend to benefit from pattern matching, maybe we should just have one UserCtx class able to answer that?